server/routes/deleteThread.ts

```diff
@@ -22,48 +28,53 @@ const deleteThread = async (models: DB, req: Request, res: Response, next: NextF
 
   try {
     const userOwnedAddressIds = (await req.user.getAddresses())
-      .filter((addr) => !!addr.verified).map((addr) => addr.id);
+      .filter((addr) => !!addr.verified)
+      .map((addr) => addr.id);
 
-    // allow either the author or admin/mods to delete threads
     const myThread = await models.OffchainThread.findOne({
       where: {
         id: req.body.thread_id,
         address_id: { [Op.in]: userOwnedAddressIds },
       },
-      include: [ models.Chain ]
+      include: [models.Chain],
     });
 
-    const thread = myThread || await models.OffchainThread.findOne({
-      where: {
-        id: req.body.thread_id,
-      },
-      include: [ models.Chain ]
-    });
+    let thread = myThread;
+    if (!myThread) {
+      const allowed_roles = {
+        siteAdmin: true,
```

Comment:
not sure why we need to specify this if we have the hardcoded `if (user.isAdmin) return true` in the validateRoles method. we only really need to know about admins + mods within the community. thusly we don't need to call options `chainAdmin` but just `admin` is fine here.